projects/fmcdaq2/src/app/fmcdaq2.c

```diff
@@ -206,77 +204,142 @@ int fmcdaq2_reconfig(struct ad9144_init_param *p_ad9144_param,
 		channel_divider = 256;
 		(&p_ad9523_param->channels[ADC_FPGA_SYSREF])->
 		channel_divider = 256;
-		p_ad9144_xcvr->reconfig_bypass = 0;
 		p_ad9144_param->lane_rate_kbps = 10000000;
-		p_ad9144_xcvr->lane_rate_kbps = 10000000;
-		p_ad9144_xcvr->ref_rate_khz = 500000;
-		p_ad9680_xcvr->reconfig_bypass = 0;
+		ad9144_xcvr_param->lane_rate_khz = 10000000;
+#ifndef ALTERA_PLATFORM
+		ad9144_xcvr_param->ref_rate_khz = 500000;
+#else
+		ad9144_xcvr_param->parent_rate_khz = 500000;
+#endif
 		p_ad9680_param->lane_rate_kbps = 5000000;
-		p_ad9680_xcvr->lane_rate_kbps = 5000000;
-		p_ad9680_xcvr->ref_rate_khz = 250000;
-#ifdef XILINX
-		p_ad9144_xcvr->dev.lpm_enable = 0;
-		p_ad9144_xcvr->dev.cpll_enable = 0;
-		p_ad9144_xcvr->dev.out_clk_sel = 4;
-
-		p_ad9680_xcvr->dev.lpm_enable = 1;
-		p_ad9680_xcvr->dev.cpll_enable = 1;
-		p_ad9680_xcvr->dev.out_clk_sel = 4;
+		ad9680_xcvr_param->lane_rate_khz = 5000000;
+#ifndef ALTERA_PLATFORM
+		ad9680_xcvr_param->ref_rate_khz = 500000;
+#else
+		ad9680_xcvr_param->parent_rate_khz = 500000;
+#endif
+#ifndef ALTERA_PLATFORM
+		ad9144_xcvr_param->lpm_enable = 0;
+		ad9144_xcvr_param->cpll_enable = 0;
+		ad9144_xcvr_param->out_clk_sel = 4;
+
+		ad9680_xcvr_param->lpm_enable = 1;
+		ad9680_xcvr_param->cpll_enable = 1;
+		ad9680_xcvr_param->out_clk_sel = 4;
 #endif
 		break;
 	default:
 		printf ("1 - ADC 1000 MSPS; DAC 1000 MSPS\n");
-		p_ad9144_xcvr->ref_rate_khz = 500000;
-		p_ad9680_xcvr->ref_rate_khz = 500000;
+#ifndef ALTERA_PLATFORM
+		ad9144_xcvr_param->ref_rate_khz = 500000;
+		ad9680_xcvr_param->ref_rate_khz = 500000;
+#else
+		ad9144_xcvr_param->parent_rate_khz = 500000;
+		ad9680_xcvr_param->parent_rate_khz = 500000;
+#endif
 		break;
 	}
 
 	return(0);
 }
 
-/***************************************************************************//**
- * @brief main
- ******************************************************************************/
 int main(void)
 {
-	int n;
+	int32_t status;
+
+	/* Initialize SPI structures */
+	struct spi_init_param ad9523_spi_param = {
+		.max_speed_hz = 2000000u,
+		.chip_select = 0,
+		.mode = SPI_MODE_0
+	};
 
-	spi_init_param	ad9523_spi_param;
-	spi_init_param	ad9144_spi_param;
-	spi_init_param	ad9680_spi_param;
+	struct spi_init_param ad9144_spi_param = {
+		.max_speed_hz = 2000000u,
+		.chip_select = 1,
+		.mode = SPI_MODE_0
+	};
+	struct spi_init_param ad9680_spi_param = {
+		.max_speed_hz = 2000000u,
+		.chip_select = 2,
+		.mode = SPI_MODE_0
+	};
 
-#ifdef ALTERA
-	ad9523_spi_param.type = NIOS_II_SPI;
-	ad9144_spi_param.type = NIOS_II_SPI;
-	ad9680_spi_param.type = NIOS_II_SPI;
+#ifndef ALTERA_PLATFORM
+	struct xil_spi_init_param xil_spi_param = {
+#ifdef PLATFORM_MB
```

Comment:
Where is `PLATFORM_MB` defined here? It would usually be in the config.h file. Is it a BSP define?

Reply:
BSP define :)